evaluator.py

```diff
@@ -55,7 +55,7 @@ def run(self, data):
                 positions: a list-like of dict-like objects.
         """
         dates = [item.date for item in data]
-        pnls = np.array([item.pnl for item in data])
+        pnls = {instrument: np.array(pnl) for instrument, pnl in data.pnl.items()}
```

Comment:
I found that this is not workable because it didn't iterate over data...
For each datum in data, datum.pnl should be a dict of str->float.
So to get pnls of a specific instrument, it should be `np.array(item.pnl[instrument] for item in data)`.
Then we have to know what instruments we have, but this is not clear in the current implementation.

To resolve this issue, @ioumu8 please add `instruments` to arguments of this method.
After that, here can be `pnls = {instrument: np.array(item.pnl[instrument] for item in data) for instrument in instruments}`.

Reply:
@hcshen916 Please remember to modify line 49 as well, let it be `def run(self, instruments, data):`

Reply:
@hcshen916 Sorry it should be `pnls = {instrument: np.array([item.pnl[instrument] for item in data]) for instrument in instruments}`. Without the pair of [], it will become a generator and so that not pickle-able.